galaxy_ng/app/api/ui/viewsets/collection.py

```diff
@@ -127,39 +129,36 @@ class CollectionVersionViewSet(api_base.GenericViewSet):
     serializer_class = serializers.CollectionVersionSerializer
 
     def list(self, request, *args, **kwargs):
-        self.paginator.init_from_request(request)
-
-        params = self.request.query_params.dict()
-        params.update({
-            'offset': self.paginator.offset,
-            'limit': self.paginator.limit,
-        })
-
-        # pulp uses ordering, but the UI is standardized around sort
-        if params.get('sort'):
-            params['ordering'] = params.get('sort')
-            del params['sort']
+        queryset = CollectionVersion.objects.all()
+        repo = self.request.query_params.dict().get('repository')
+        if repo:
+            try:
+                distro = AnsibleDistribution.objects.get(base_path=repo)
+            except ObjectDoesNotExist:
+                raise NotFound(f'Repository not found: {repo}')
+            repository_version = distro.repository.latest_version()
+            queryset = queryset.filter(pk__in=repository_version.content)
```

Comment:
Could / should this be a FilterSet ?

Reply:
I'll look at that

Reply:
It just occurred to me that since _ui/collections/ and v3/collections/ are both views to the same Collection models, if we add any Collection view FilterSets we could potentially use them in both places.

Reply:
(Not necessarily a requirement now though...)

Reply:
Added a FilterSet.

For sharing across `v3` and `_ui`... at least for CollectionVersion there are some differences: `v3` filters to just certified vs not in `_ui`, and `v3` uses url kwargs for namespace and name vs query params for those in `_ui`.